classes/observation_manager.php

```diff
@@ -0,0 +1,402 @@
+
+        // Get all the points to get the bounds.
+        $allpoints = self::get_observation_points($observationid);
+        $alllistorders = array_column($allpoints, 'list_order');
+
+        if (count($alllistorders) === 0) {
+            throw new \coding_exception('No list orderings found for this observation instance, but expected at least one.');
+        }
+
+        $newordering = $targetpoint->list_order + $direction;
+
+        // Is currently the minimum - do nothing.
+        if ($newordering < min($alllistorders)) {
+            return;
+        }
+
+        // Is currently the maximum - do nothing.
+        if ($newordering > max($alllistorders)) {
+            return;
+        }
+
+        // Ordering is valid, so get the points that are affected by this reordering.
+        $affectedpoints = array_filter($allpoints, function($elem) use($newordering, $direction, $targetpoint) {
+            // Don't include the item being reordered.
+            if ($elem->id === $targetpoint->id) {
+                return false;
+            }
+
+            // If shifting down, filter those 'above'.
+            if ($direction < 0) {
+                return $elem->list_order >= $newordering && $elem->list_order < $targetpoint->list_order;
+            }
+            // If shifting up, filter those 'below'.
+            if ($direction > 0) {
+                return $elem->list_order <= $newordering && $elem->list_order > $targetpoint->list_order;
+            }
+
+            return false;
+        });
+
+        $transaction = $DB->start_delegated_transaction();
+
+        // Give the target point the new ordering.
+        $DB->update_record($tablename,
+        [
+            'id' => $targetpoint->id,
+            'list_order' => $newordering
+        ]);
+
+        // Reduce the direction to a unit vector (e.g. 5 -> 1 and -5 -> -1).
+        $reductionamount = intdiv($direction, abs($direction));
+
+        array_walk($affectedpoints, function($elem) use($DB, $tablename, $reductionamount) {
+            $DB->update_record($tablename,
+            [
+                'id' => $elem->id,
+                'list_order' => $elem->list_order - $reductionamount
+            ]);
+        });
+
+        $transaction->allow_commit();
+    }
+
+    /**
+     * Returns a list of observation points with responses for a particular session
+     * @param int $observationid ID of the observation instance
+     * @param int $sessionid ID of the observation sesssion
+     * @return array array of observation points with responses for the given session
+     */
+    public static function get_points_and_responses(int $observationid, int $sessionid) {
+        global $DB;
+
+        // Selects all points for this observation,
+        // and attaches responses for the current session (if one exists).
+
+        $sql = 'SELECT pts.id as point_id, obs_id, title, list_order, ins, ins_f, max_grade, res_type,
```

Comment:
any blocker to simplifying this to * and just ignoring whatever fields you dont need?